module/VuFind/src/VuFind/Config/Handler/Ini.php

```diff
@@ -130,4 +140,143 @@ protected function getMergeCallback(array $overrideFullSections, bool $mergeArra
         };
     }
+
+    /**
+     * Write configuration to a specific location.
+     *
+     * @param ConfigLocationInterface  $destinationLocation Destination location for the config
+     * @param array|string             $config              Config to write
+     * @param ?ConfigLocationInterface $baseLocation        Location of a base configuration that can provide additional
+     * structure (e.g. comments)
+     *
+     * @return void
+     */
+    public function writeConfig(
+        ConfigLocationInterface $destinationLocation,
+        array|string $config,
+        ?ConfigLocationInterface $baseLocation
+    ): void {
+        if (!is_array($config)) {
+            throw new ConfigException('Ini handler can only write array config.');
+        }
+
+        // If target file already exists, back it up:
+        $outfile = $destinationLocation->getPath();
+        $this->backupFile($outfile);
+
+        $comments = [];
+        if ($baseLocation !== null) {
+            $comments = $this->extractComments($baseLocation->getPath());
+        }
+        $writer = new ConfigWriter(
+            $outfile,
+            $config,
+            $comments
+        );
+        if (!$writer->save()) {
+            throw new FileAccessException(
+                "Error: Problem writing to {$outfile}."
+            );
+        }
+    }
+
+    /**
+     * Read the specified file and return an associative array of this format
+     * containing all comments extracted from the file:
+     *
+     * [
+     *   'sections' => array
+     *     'section_name_1' => array
+     *       'before' => string ("Comments found at the beginning of this section")
+     *       'inline' => string ("Comments found at the end of the section's line")
+     *       'settings' => array
+     *         'setting_name_1' => array
+     *           'before' => string ("Comments found before this setting")
+     *           'inline' => string ("Comments found at the end of setting's line")
+     *           ...
+     *         'setting_name_n' => array (same keys as setting_name_1)
+     *        ...
+     *      'section_name_n' => array (same keys as section_name_1)
+     *   'after' => string ("Comments found at the very end of the file")
+     * ]
+     *
+     * @param string $filename Name of ini file to read.
+     *
+     * @return array           Associative array as described above.
+     */
+    protected function extractComments($filename)
+    {
+        $lines = file($filename);
+
+        // Initialize our return value:
+        $retVal = ['sections' => []];
+
+        // Initialize variables for tracking status during parsing:
+        $section = $comments = '';
+
+        foreach ($lines as $line) {
+            // To avoid redundant processing, create a trimmed version of the current
+            // line:
+            $trimmed = trim($line);
+
+            // Is the current line a comment?  If so, add to the currentComments
```

Comment:
This comment seems to refer to a variable name that no longer exists:

```suggestion
            // Is the current line a comment?  If so, add to the current comments
```